ext/web/06_streams.js

```diff
@@ -742,14 +742,15 @@ function readableStreamForRid(rid, autoClose = true) {
           return;
         }
 
-        const bytesRead = await core.read(rid, v);
+        // deno-lint-ignore prefer-primordials
+        const [bytesRead, newView] = await core.readDetaching(rid, v);
```

Comment:
How is this correct? If a BYOB read occurs, the view shouldn't get detached for the duration of this read, right?

The view is still accessible to user code. Also, if the identity of the view is not preserved, why support BYOB in the first place?

Reply:
`ReadableStreamBYOBReader.prototype.read` detaches the given buffer ([`ReadableByteStreamControllerPullInto`](https://streams.spec.whatwg.org/#readable-byte-stream-controller-pull-into) step 7) and returns a new one with the same backing store. When you implement the underlying source, you can use `ReadableStreamBYOBRequest.prototype.respondWithNewView` to respond with a new buffer that points to the same backing store.